### DISCUSSION TITLE
[Bug]: Must click multiple times on toggle button before state updates using ToggleButtonGroup and ToggleButton from Material UI -

### DISCUSSION THREAD

**Discussion by idelice** [react-hook-form/react-hook-form#6261] [Q&A] [answered]
### Version Number

7.12.2

### Codesandbox/Expo snack

https://codesandbox.io/s/trusting-shadow-3bjzs?file=/src/App.tsx

### Steps to reproduce

1. Go to the only page there is
2. Click on "Junior" or "Senior"
3. Error: state never changes on first click. You must continue to click on either one till one of the changes state.

### Expected behaviour

I expect to only click **_once_** for the state to change.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by idelice** [comment]
Here is the problem visualised: 
![rhfMui](https://user-images.githubusercontent.com/59118341/129918359-7ee51281-8065-4a69-9ee1-c87fa92f7ecd.gif)

**Comment by bluebill1049** [comment, marked as answer]
check mui doc for onChange handler for that component, it's probably `(e, value) => field.onChange(value)`

**Reply by idelice** [reply]
Yes, you are correct. I must have used onChange prop like: `onChange={(e, value) => onChange(value)}`.

Thank you.
